datalad/support/repodates.py

```diff
@@ -0,0 +1,252 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Utilities for checking repository dates.
+"""
+
+import logging
+import operator
+import re
+import time
+
+from six import string_types
+
+from datalad.support.gitrepo import GitRepo, GitCommandError
+
+lgr = logging.getLogger('datalad.repodates')
+
+
+def branch_blobs(repo, branch):
+    """Get all blobs for `branch`.
+
+    Parameters
+    ----------
+    repo : GitRepo
+    branch : str
+
+    Returns
+    -------
+    A generator object that returns (hexsha, content, file name) for each blob
+    in `branch`.  Note: By design a blob isn't tied to a particular file name;
+    the returned file name matches what is returned by 'git rev-list'.
+    """
+    git = repo.repo.git
+    # Note: This might be nicer with rev-list's --filter and
+    # --filter-print-omitted, but those aren't available until Git v2.16.
+    lines = git.rev_list(branch, objects=True).splitlines()
+    # Trees and blobs have an associated path printed.
+    objects = (ln.split() for ln in lines)
+    blob_trees = (obj for obj in objects if len(obj) == 2)
+
+    # This is inefficient.  It makes a git call for each object, some of which
+    # aren't even blobs.  We could instead use 'git cat-file --batch'.
+    for obj, fname in blob_trees:
+        try:
+            yield obj, git.cat_file("blob", obj), fname
+        except GitCommandError:  # The object was a tree.
+            continue
```

Comment:
so I guess you didn't see it  easy to implement via cat-file?  or isn't it already implemented (I see above cat_file) and the comment is outdated? or it refers to `blob_trees` generator?

Reply:
> so I guess you didn't see it easy to implement via cat-file?  or isn't it already implemented (I see above cat_file) and the comment is outdated?

What is "it"?  (Briefly looking at the comments in `branch_blobs`, none seem outdated.)

> or it refers to blob_trees generator?

Sorry, I'm not having any luck parsing this either.